### PR TITLE
fix: Select component inconsistently render selection styles in dropdown 

### LINKED ISSUE #21768: Select component implementations inconsistently render selection styles in dropdown
## **Summary**
Our Select component is used in a few places:

* [SelectCloudProjectModal.vue](https://github.com/cypress-io/cypress/blob/92eac2f/packages/app/src/runs/modals/SelectCloudProjectModal.vue#L11)
* [ChooseExternalEditor.vue](https://github.com/cypress-io/cypress/blob/92eac2f67e4a0391bf058f8f95b64cc731d609cf/packages/frontend-shared/src/gql-components/ChooseExternalEditor.vue#L3)
* [SelectFwOrBundler.vue](https://github.com/cypress-io/cypress/blob/92eac2f67e4a0391bf058f8f95b64cc731d609cf/packages/launchpad/src/setup/SelectFwOrBundler.vue#L2)

Most do not show the current value as selected in the dropdown with the visible checkmark and other styling, with the exception of the "Front-end Framework" dropdown in SelectFwOrBundler. However, it seems like they all should show these selection styles.

The Select currently expects the selected options to be provided with an `isSelected` property, which is not uniformly provided today. It seems like we could use the inner [TabList's](https://headlessui.dev/vue/listbox#styling-the-active-and-selected-option) selection APIs to keep this synchronized better.

## **Acceptance Criteria**
* Select implementations within these files should consistently show their selected items as selected in the dropdown
* Percy snapshots should be added for these scenarios

Screen Shot 2022-03-17 at 5 00 09 PM
Screen Shot 2022-03-17 at 5 17 04 PM
Screen Shot 2022-03-17 at 5 07 38 PM
Screen Shot 2022-03-17 at 5 07 22 PM
Screen Shot 2022-03-17 at 5 00 25 PM

┆Issue is synchronized with this [Jira Bug](https://cypress-io.atlassian.net/browse/UNIFY-1375) by [Unito](https://www.unito.io)
┆Attachments: <a href="https://cypress-io.atlassian.net/rest/api/2/attachment/content/11525">Screen Shot 2022-03-17 at 5.00.09 PM.png</a> | <a href="https://cypress-io.atlassian.net/rest/api/2/attachment/content/11524">Screen Shot 2022-03-17 at 5.00.25 PM.png</a> | <a href="https://cypress-io.atlassian.net/rest/api/2/attachment/content/11528">Screen Shot 2022-03-17 at 5.07.22 PM.png</a> | <a href="https://cypress-io.atlassian.net/rest/api/2/attachment/content/11527">Screen Shot 2022-03-17 at 5.07.38 PM.png</a> | <a href="https://cypress-io.atlassian.net/rest/api/2/attachment/content/11526">Screen Shot 2022-03-17 at 5.17.04 PM.png</a>
┆author: Tyler Biethman
┆epic: UI/UX Improvements
┆friendlyId: UNIFY-1375
┆priority: Low
┆sprint: Backlog
┆taskType: Bug

### PR DESCRIPTION
**Pull request by gitstart**
- Closes https://github.com/cypress-io/cypress/issues/21768

### User facing changelog

Fix issue for the `Select Component` do not show the current value as selected in the dropdown with the visible checkmark and other styling. 

### Additional details

The change is necessary to improve accessibility by clearly indicating the selected `List Item` from the `Dropdown Menu`.

### Steps to test

1. Run the command `yarn dev`
2. Add a new project that hasn't been configured with  `Cypress Test` before.
3. For every dropdown menu, the selected `List Item` should have a `bg-jade-50` background and `tick` checkIcon.

#### Before
https://www.loom.com/share/eac3bfa96d1744eaa14a0adec8d1b357

#### After
https://www.loom.com/share/e5291eba62dd4303a7fcb22783ff465c

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by gitstart** [comment]
Hi @marktnoonan could you help to make pipeline run on this branch?

**Comment by gitstart** [comment]
Hi @astone123 @AtofStryker kindly help to take a look at this PR

cc: @lmiller1990

File: packages/frontend-shared/src/components/Select.cy.tsx [hunk 1]
```diff
@@ -101,13 +101,13 @@ describe('<Select />', () => {
   })
 
   describe('#icons', () => {
-    // TODO: Enable with completion of UNIFY-1375
```
Comment (AtofStryker) [inline packages/frontend-shared/src/components/Select.cy.tsx:104]: @marktnoonan @warrensplayer can we close this issue now or is it already closed?

**Review by AtofStryker — APPROVED** [review]

**Comment by mschile** [comment]
@gitstart, it doesn't look like the 'External Editor' selection or 'Organization/Project' selections show the highlighted selection with check.
Screen Shot 2022-10-28 at 8 54 30 AM
Screen Shot 2022-10-28 at 8 54 48 AM

File: packages/frontend-shared/src/components/Select.vue [hunk 2]
```diff
@@ -177,6 +183,12 @@ const props = withDefaults(defineProps<{
   itemKey: 'key',
 })
 
+const isSelectedOption = (option: Option) => {
+  const optionKey = get(option, props.itemKey)
+
+  return optionKey && optionKey === get(props.modelValue, props.itemKey)
+}
+
```
Comment (gitstart) [inline packages/frontend-shared/src/components/Select.vue:191]: @mschile Thanks for showing us the issue. we found out that we are not keeping ref between `modelValue` and `option` in usages, to resolve it we use `itemKey` to check if an option is selected. WDYT about this way?

File: packages/frontend-shared/src/components/Select.vue [hunk 3]
```diff
@@ -166,7 +166,13 @@ const props = withDefaults(defineProps<{
   modelValue?: Option // Current object being selected
   placeholder?: string
   label?: string
-  itemValue?: string // The key of the modelValue to render
+  /**
+   * The key of the modelValue to render. `value` by default
```
Comment (mschile) [inline packages/frontend-shared/src/components/Select.vue]:
```suggestion
   * The value of the modelValue to render. `value` by default
```

**Review by mschile — APPROVED** [review]
LGTM. Verified the selection styles in the "Project Setup" for component testing, the "External Editor" in device settings, and the "Connect project" modal.

**Review by AtofStryker — APPROVED** [review]

**Comment by mschile** [comment]
Thanks for your contribution @gitstart!

**Comment by raph941** [comment]
This PR was pushed through [Gitstart](https://www.gitstart.com/), with contributions from @oxenprogrammer, @raph941, @phunguyenmurcul.

**Comment by gitstart** [comment]
This PR was pushed through [Gitstart](https://www.gitstart.com/), with contributions from @oxenprogrammer, @raph941, @phunguyenmurcul.

**Pull request merged**